src/main/java/com/mraof/minestuck/util/StartingModusManager.java

```diff
@@ -0,0 +1,56 @@
+package com.mraof.minestuck.util;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.reflect.TypeToken;
+import com.mraof.minestuck.Minestuck;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.ResourceManager;
+import net.minecraft.server.packs.resources.SimpleJsonResourceReloadListener;
+import net.minecraft.util.profiling.ProfilerFiller;
+import net.minecraftforge.event.AddReloadListenerEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.FORGE)
+public class StartingModusManager extends SimpleJsonResourceReloadListener
+{
+	private static final Gson GSON = new GsonBuilder().registerTypeAdapter(BoondollarPricing.class, new BoondollarPricing.Serializer()).create();
+	
+	private static StartingModusManager INSTANCE;
+	private List<String> startingModusTypes;
```

Comment:
One improvement that should now be easy to make is to introduce weights. So letting each modus type to be paired with a weight number that determines how ofter that modus type is chosen.

Consider this a soft suggestion, as I don't think "weights" is a particularly important feature in this specific case.

Reply:
I dont see the immediate need for it right now